lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/SplitBamByReadGroup.pm

```diff
@@ -93,17 +93,17 @@ sub _process_reads {
         }
 
         if($tokens[0] eq $previous_tokens->[0] and $previous_read_group_id eq $read_group_id) {
-            $self->_write_reads_based_on_read_group_and_pairedness(
+            $self->_write_reads_based_on_read_group_and_type(
                 rg_id => $read_group_id,
-                pairedness => 'paired',
+                type => 'paired',
                 reads => [ $previous_tokens, \@tokens ],
             );
             undef $previous_tokens;
             undef $previous_read_group_id;
         } else {
-            $self->_write_reads_based_on_read_group_and_pairedness(
+            $self->_write_reads_based_on_read_group_and_type(
                 rg_id => $previous_read_group_id,
-                pairedness => 'singleton',
+                type => ( $previous_tokens->[1] & 0x80 ? 'read2' : 'read1' ), # unlabeled reads in read1
```

Comment:
It might be nice to encapsulate this type-decision logic, both to eliminate the duplication and to give a name to the otherwise obscure `&`ing.